Name: No Auto-Off While Charging
Version: 1.4.5-1
Author: Jason Robitaille
Description: Prevents auto-off while charging  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1868,6 +1868,8 @@
 			if (this.powerSource[source] == true)
 				this.isCharging = true;			
 		}.bind(this));
+		
+		this.setForAutoOff();
 						
 		if (this.isCharging) {
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
@@ -1901,6 +1903,56 @@
 	}
 },
 
+setForAutoOff: function() {
+	if(this.isCharging==true) {
+		this.getAutoOffTimeout(
+			function(response) {
+				if(response.timeout!=604800) {
+					this.setAutoOffCookie(response.timeout);
+					this.setAutoOffTimeout(604800);
+				}
+			}.bind(this)
+		);
+	} else {
+		var timeout = this.getAutoOffCookie();
+		if(timeout.seconds != 604800) {
+			this.setAutoOffTimeout(timeout.seconds);
+			this.setAutoOffCookie(604800);
+		}
+	}
+},
+
+getAutoOffCookie: function() {
+	var cookie = new Mojo.Model.Cookie("auto-off");
+	var val = cookie.get();
+	if(!val) {
+		val = {seconds:604800};
+		cookie.put(val);
+	}
+	return val;
+},
+
+setAutoOffCookie: function(value) {
+	var cookie = new Mojo.Model.Cookie("auto-off");
+	cookie.put({seconds:value});
+},
+
+getAutoOffTimeout: function(callback) {
+	this.controller.serviceRequest('palm://com.palm.display/control', {
+		method: 'getProperty',
+		parameters:{properties:['timeout']},
+		onSuccess: callback
+	});
+},
+
+setAutoOffTimeout: function(value) {
+	this.controller.serviceRequest('palm://com.palm.display/control', {
+		method: 'setProperty',
+		parameters:{timeout:value}
+	});
+
+},
+
 handleCriticalPowerNotifications: function(payload) {
 	if(payload.cause) {
 		Mojo.Log.error("SystemUI - Battery Level is critical. Shutting down the phone ");
